src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestRetryFailedManagedIndex.kt

```diff
@@ -0,0 +1,168 @@
+                                if (acknowledgedResponse.isAcknowledged) {
+                                    builder.field(UPDATED_INDICES, listOfIndexMetadata.size)
+                                } else {
+                                    failedIndices.addAll(listOfIndexMetadata.map {
+                                        FailedIndex(it.first.name, it.first.uuid, "failed to update IndexMetadata")
+                                    })
+                                }
+                                buildInvalidIndexResponse(builder)
+                                return BytesRestResponse(RestStatus.OK, builder.endObject())
+                            }
+                        }
+                    )
+                } catch (e: ClusterBlockException) {
+                    failedIndices.addAll(listOfIndexMetadata.map {
+                        FailedIndex(it.first.name, it.first.uuid, "failed to update with ClusterBlockException. ${e.message}")
+                    })
+                    buildInvalidIndexResponse(builder)
+                    channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
+                }
+            } else {
+                buildInvalidIndexResponse(builder)
+                channel.sendResponse(BytesRestResponse(RestStatus.OK, builder.endObject()))
+            }
+        }
+
+        private fun buildInvalidIndexResponse(builder: XContentBuilder) {
+            if (failedIndices.isNotEmpty()) {
+                builder.field(FAILURES, true)
+                builder.startArray(FAILED_INDICES)
+                for (failedIndex in failedIndices) {
+                    builder.startObject()
+                    builder.field("index_name", failedIndex.name)
+                    builder.field("index_uuid", failedIndex.uuid)
+                    builder.field("reason", failedIndex.reason)
+                    builder.endObject()
+                }
+                builder.endArray()
+            } else {
+                builder.field(FAILURES, false)
+            }
+        }
+
+        private fun populateList(state: ClusterState, startState: String?) {
+            for (indexMetadataEntry in state.metaData.indices) {
+                val indexMetadata = indexMetadataEntry.value
+
+                if (indexMetadata.getPolicyName() == null) {
+                    failedIndices.add(FailedIndex(indexMetadata.index.name, indexMetadata.index.uuid, "This index is not being managed."))
+                    continue
+                }
+
+                val managedIndexMetaData = indexMetadata.getManagedIndexMetaData()
+                if (managedIndexMetaData == null) {
+                    failedIndices.add(FailedIndex(indexMetadata.index.name, indexMetadata.index.uuid, "There is no index metadata information"))
+                } else {
+                    if (!isFailedState(managedIndexMetaData)) {
```

Comment:
np: Not the biggest fan of `isFailedState` that just checks if metadata.failed is true. I'd prefer one of:
a) Just check `if (managedIndexMetaData.failed != true) { ... }`
b) Extension property `managedIndexMetaData.isFailedState`